godtools/Views/TractElements/BaseTractElement.swift

```diff
@@ -143,16 +143,29 @@ class BaseTractElement: UIView {
         var currentYPosition: CGFloat = 0.0
         var maxYPosition: CGFloat = 0.0
         var elements = [BaseTractElement]()
+        var previousElementWasImage = false
+        var previousPosition: CGFloat = 0.0
         
         for dictionary in data {
-            let element = buildElementForDictionary(dictionary, startOnY: currentYPosition)
-            if self.horizontalContainer && element.yEndPosition() > maxYPosition {
-                maxYPosition = element.yEndPosition()
+            var element: BaseTractElement?
+            let properties = splitData(data: dictionary)
+            
+            if previousElementWasImage && properties.kind == "image" {
```

Comment:
Can you explain a little bit what you're doing in this section? I am not following it yet.